presto-hive/src/main/java/com/facebook/presto/hive/s3/PrestoS3FileSystem.java

```diff
@@ -854,8 +855,23 @@ private AWSCredentialsProvider createAwsCredentialsProvider(URI uri, Configurati
             return InstanceProfileCredentialsProvider.getInstance();
         }
 
-        if (s3IamRole != null) {
-            return new STSAssumeRoleSessionCredentialsProvider.Builder(s3IamRole, s3IamRoleSessionName).build();
+        boolean webIdentityEnabled = conf.getBoolean("presto.hive.s3.webidentity.enabled", false);
+        String webIdentityTokenFile = conf.get("presto.hive.s3.web-identity-token-file");
+
+        if (!isNullOrEmpty(s3IamRole)) {
+            if (webIdentityEnabled) {
+                log.info("Using Web Identity Token Credentials Provider.");
```

Comment:
nit: We should minimize info logs in production and use them only when absolutely necessary. If logging is required in this case, we can add a debug log.